tests/test_apply.py

```diff
@@ -243,6 +243,33 @@ def test_dry_run(monkeypatch):
         s3.head_bucket(Bucket=bucket_name)
 
 
+def test_service_endpoint_alias_replacements(monkeypatch):
+    monkeypatch.setenv("DRY_RUN", "1")
+    config = """
+    provider "aws" {
+      region = "eu-west-1"
+    }"""
+
+    temp_dir = deploy_tf_script(config, cleanup=False, user_input="yes")
+    override_file = os.path.join(temp_dir, "localstack_providers_override.tf")
+    assert check_override_file_content(override_file)
+    rmtree(temp_dir)
+
+
+def check_override_file_content(override_file):
+    try:
+        with open(override_file, "r") as fp:
+            result = hcl2.load(fp)
+            result = result["provider"][0]["aws"]
+    except Exception as e:
+        print(f'Unable to parse "{override_file}" as HCL file: {e}')
+
+    endpoints = result["endpoints"][0]
+    if "config" in endpoints and "configservice" in endpoints:
```

Comment:
Instead of hardcoding the service names here, could we pass in a list of expected services as a parameter of the function?

Also, I would have expected this to be something like this - wouldn't we expect only one of the two be contained in the `endpoints` list (and not both)..? 🤔 
```
if "config" not in endpoints and "configservice" in endpoints:
```

Reply:
Sure, not a problem to use them es parameters. 
However I'd argue the logic change here:
If the list order changes in tflocal that breaks the test for no reason, and not necessarily wrong to use `configservice` instead of `config` or vice versa as it's a valid option, we just want to use exclusively one in any given point to hide the warnings, so we made the decision to do it with the first item from the aliases in tflocal.

Or do you think we should focus more here to test the actual values we should see in the generated file?

Reply:
Ok. I think the logic makes sense, thanks for clarifying. Maybe we could still make the logic parametrizable, rather than harcoding the service names. 

Btw, if I understand correctly, we want to ensure that no duplicates are being generate - could we simply iterate over all alias pairs within `SERVICE_ALIASES`, and ensure that only one of the aliases is defined in the config..? 🤔

Reply:
That's already happenning here :)
https://github.com/localstack/terraform-local/blob/6d9f75b3b3c1cafd1141db04283a66fbf5d49f04/bin/tflocal#L150
Basically we iterate through all aliases, if the alias is among the service endpoints we add the service endpoint by the first alias name so in case we find it multiple times we simply just updating the service endpoint dict at the respective key.

Reply:
Yep, that makes sense, great to see that we have this logic in place for `create_provider_config_file`. 👍  I was more referring to the test assertion here, if we are deliberately checking for `config`/`configservice` only. But if we're happy with the assertion here, I'm fine with merging as-is (we can evolve this as we go.. ➿ ).